pmd-java/src/main/java/net/sourceforge/pmd/lang/java/oom/Metrics.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.oom;
+
+import static net.sourceforge.pmd.lang.java.oom.visitor.MetricsVisitorFacade.TOP_LEVEL_PACKAGE;
+
+import net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceDeclaration;
+import net.sourceforge.pmd.lang.java.ast.ASTMethodOrConstructorDeclaration;
+import net.sourceforge.pmd.lang.java.ast.QualifiableNode.QualifiedName;
+import net.sourceforge.pmd.lang.java.oom.metrics.AtfdMetric;
+import net.sourceforge.pmd.lang.java.oom.metrics.WmcMetric;
+
+
+/**
+ * User bound façade of the Metrics Framework. Provides a uniform interface for the calculation of
+ * metrics.
+ *
+ * @author Clément Fournier
+ */
+public class Metrics {
+
+    private Metrics() { // Cannot be instantiated
+
+    }
+
+    /**
+     * Computes a metric identified by its code on the class AST node being
+     * passed.
+     */
+    public static double get(ClassMetricKey key, ASTClassOrInterfaceDeclaration node) {
+        QualifiedName qname = node.getQualifiedName();
+        double memoized = TOP_LEVEL_PACKAGE.getMemo(key, qname); // TODO
+
+        return memoized == Double.NaN ? key.getCalculator().computeFor(node, TOP_LEVEL_PACKAGE) : memoized;
```

Comment:
In order to determine, whether a metric has already been calculated/memoized, you could also use the `Double` type and check against `null` instead of NaN.